assets/chat/js/menus.js

```diff
@@ -168,31 +169,41 @@ class ChatSettingsMenu extends ChatMenu {
     }
 
     updateNotification(){
-        const perm = Notification.permission === 'default' ? 'required' : Notification.permission
-        this.notificationEl.text(`(Permission ${perm})`)
+        if( Notification.permission === "granted" ) {
+            this.enableNotificationsBtn.text("Permission granted ✓");
+            this.enableNotificationsBtn.attr('disabled', true);
+            this.notificationsFieldset.attr('disabled', false);
+        } else {
+            this.enableNotificationsBtn.text("Enable Notifications");
+            this.enableNotificationsBtn.attr('disabled', false);
+            this.notificationsFieldset.attr('disabled', true);
+        }
     }
 
     notificationPermission(){
         return new Promise((resolve, reject) => {
-            switch(Notification.permission) {
-                case 'default':
-                    Notification.requestPermission(permission => {
-                        switch(permission) {
-                            case 'granted':
-                                resolve(permission);
-                                break;
-                            default:
-                                reject(permission);
-                        }
-                    });
-                    break;
-                case 'granted':
-                    resolve(Notification.permission);
-                    break;
-                case 'denied':
-                default:
-                    reject(Notification.permission);
-                    break;
+            // Detecting for Blink webengine and whether we are not the top frame
+            // See issue github.com/MemeLabs/chat-gui/issues/12
+            const isBlink = window.Intl && Intl.v8BreakIterator;
+            const isEmbedded = window.top !== self;
+
+            if( (isEmbedded && isBlink) || Notification.permission === "denied" ) {
+                window.open(
+                    "notification-request.html",
+                    "strimsgg_notification_request",
+                    "width=585,height=340,scrollbars=no,toolbar=no"
+                );
+
+                const onMessage = event => {
+                    if( event.data.name === "notification-request-done" ){
+                        window.removeEventListener('message', onMessage);
+                        resolve(Notification.permission);
+                    }
+                };
+
+                window.addEventListener( 'message', onMessage );
+            } else {
+                Notification.requestPermission( resolve );
```

Comment:
Other function calls in this file don't add spaces before and after the list of parameters.